src/core/Akka.Remote.TestKit/MsgDecoder.cs

```diff
@@ -54,9 +54,9 @@ protected object Decode(object message)
                     {
                         case Proto.Msg.EnterBarrier.Types.BarrierOp.Succeeded: return new BarrierResult(w.Barrier.Name, true);
                         case Proto.Msg.EnterBarrier.Types.BarrierOp.Failed: return new BarrierResult(w.Barrier.Name, false);
-                        case Proto.Msg.EnterBarrier.Types.BarrierOp.Fail: return new FailBarrier(w.Barrier.Name);
+                        case Proto.Msg.EnterBarrier.Types.BarrierOp.Fail: return new FailBarrier(w.Barrier.Name, new RoleName(w.Barrier.RoleName));
```

Comment:
Added wire format support for the new `RoleName` values to both `FailBarrier` and `EnterBarrier` on the decoding side.